SYCL/GroupAlgorithm/SYCL2020/exclusive_scan.cpp

```diff
@@ -37,18 +37,20 @@ OutputIterator exclusive_scan(InputIterator first, InputIterator last,
 }
 } // namespace emu
 
-template <typename SpecializationKernelName, typename InputContainer,
-          typename OutputContainer, class BinaryOperation>
+template <typename T> bool approx_eq(T a, T b) {
+  return std::abs(a - b) / (0.5 * std::abs(a + b)) < 1e-6;
+}
+
```

Comment:
Is this really necessary?  Could we instead use inputs that guarantee the result can be represented exactly by a float?

I'm worried that these sorts of approximations could hide bugs...  For example, if the group algorithms incorrectly set an accumulator to something very small, wouldn't this test still pass?